File: ee25btech11047/Matgeo/problem4/codes/2.8.37.py
import numpy as np

# Given values
a = 4.0        # |a|
lhs = 144.0    # |a × b|^2 + (a · b)^2

# Using numpy
b_squared = lhs / np.power(a, 2)
b = np.sqrt(b_squared)

print(f"|b| = {b}")

File: ee25btech11047/Matgeo/problem4/codes/2.8.37.c
#include <stdio.h>
#include <math.h>

int main() {
    // Given values
    double a = 4.0;   // |a| = 4
    double lhs = 144; // |a × b|^2 + (a · b)^2 = 144

    // From identity: |a × b|^2 + (a · b)^2 = |a|^2 * |b|^2
    // => |b|^2 = lhs / (|a|^2)
    double b_squared = lhs / (a * a);
    double b = sqrt(b_squared);

    printf("|b| = %.2f\n", b);

    return 0;
}
